src/dev/flang/be/interpreter2/Intrinsics.java

```diff
@@ -0,0 +1,1534 @@
+        {
+          var d = args.get(1).f64Value();
+          var jd = Double.valueOf(d);
+          Clazz resultClazz = innerClazz.resultClazz();
+          return JavaInterface.javaObjectToInstance(jd, resultClazz);
+        });
+    putUnsafe("fuzion.java.bool_to_java_object", (interpreter, innerClazz) -> args ->
+        {
+          var b = args.get(1).boolValue();
+          var jb = Boolean.valueOf(b);
+          Clazz resultClazz = innerClazz.resultClazz();
+          return JavaInterface.javaObjectToInstance(jb, resultClazz);
+        });
+    put("fuzion.sys.internal_array_init.alloc", (interpreter, innerClazz) -> args ->
+        {
+          return ArrayData.alloc(/* size */ args.get(1).i32Value(),
+                                     /* type */ innerClazz._outer);
+        });
+    put("fuzion.sys.internal_array.get", (interpreter, innerClazz) -> args ->
+        {
+          return ((ArrayData)args.get(1)).get(
+                                   /* index */ args.get(2).i32Value(),
+                                   /* type  */ ArrayData.elementType(innerClazz._outer));
+        });
+    put("fuzion.sys.internal_array.setel", (interpreter, innerClazz) -> args ->
+        {
+          ((ArrayData)args.get(1)).set(
+                              /* index */ args.get(2).i32Value(),
+                              /* value */ args.get(3),
+                              /* type  */ ArrayData.elementType(innerClazz._outer));
+          return Value.EMPTY_VALUE;
+        });
+    put("fuzion.sys.internal_array.freeze", (interpreter, innerClazz) -> args ->
+        {
+          return Value.EMPTY_VALUE;
+        });
+    put("fuzion.sys.internal_array.ensure_not_frozen", (interpreter, innerClazz) -> args ->
+        {
+          return Value.EMPTY_VALUE;
+        });
+    put("fuzion.sys.env_vars.has0", (interpreter, innerClazz) -> args -> new boolValue(System.getenv(utf8ByteArrayDataToString(args.get(1))) != null));
+    put("fuzion.sys.env_vars.get0", (interpreter, innerClazz) -> args -> Interpreter2.value(System.getenv(utf8ByteArrayDataToString(args.get(1)))));
+    // setting env variable not supported in java
+    put("fuzion.sys.env_vars.set0"  , (interpreter, innerClazz) -> args -> new boolValue(false));
+    // unsetting env variable not supported in java
+    put("fuzion.sys.env_vars.unset0", (interpreter, innerClazz) -> args -> new boolValue(false));
+    put("fuzion.sys.misc.unique_id",(interpreter, innerClazz) -> args -> new u64Value(_last_unique_id_.incrementAndGet()));
+    put("fuzion.sys.thread.spawn0", (interpreter, innerClazz) -> args ->
+        {
+          var call = Types.resolved.f_function_call;
+          var oc = innerClazz.argumentFields()[0].resultClazz();
+          var ic = oc.lookup(call);
+          var t = new Thread(() -> interpreter.callOnInstance(ic._idInFUIR, new Instance(ic), args.get(1), new List<>(), false /* NYI could we have pre-cond here? */));
```

Comment:
`Function.call` does not (and should never) have a pre-condition, I think the comment can go.